### PR TITLE
feat: new type varbinary

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

**Comment by sangshuduo** [comment]
https://github.com/taosdata/TDengine/issues/22795

File: src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java [hunk 1]
```diff
@@ -482,7 +486,8 @@ public long getLong(int col) throws SQLException {
                 return Long.parseLong((String) obj);
             }
             case TSDBConstants.TSDB_DATA_TYPE_JSON:
-            case TSDBConstants.TSDB_DATA_TYPE_BINARY: {
+            case TSDBConstants.TSDB_DATA_TYPE_BINARY:
+            case TSDB_DATA_TYPE_VARBINARY: {
```
Comment (sangshuduo) [inline src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java:490]: not consistent as before

File: src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java [hunk 2]
```diff
@@ -568,7 +582,8 @@ public double getDouble(int col) throws SQLException {
                 return Double.parseDouble((String) obj);
             }
             case TSDBConstants.TSDB_DATA_TYPE_JSON:
-            case TSDBConstants.TSDB_DATA_TYPE_BINARY: {
+            case TSDBConstants.TSDB_DATA_TYPE_BINARY:
+            case TSDB_DATA_TYPE_VARBINARY: {
```
Comment (sangshuduo) [inline src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java:586]: same comment

File: src/main/java/com/taosdata/jdbc/TSDBResultSetRowData.java [hunk 3]
```diff
@@ -91,7 +91,8 @@ public boolean getBoolean(int col, int nativeType) throws SQLException {
                 return parseUBigInt((long) obj).equals(new BigDecimal(0)) ? Boolean.FALSE : Boolean.TRUE;
             case TSDBConstants.TSDB_DATA_TYPE_BINARY:
             case TSDBConstants.TSDB_DATA_TYPE_JSON:
-            case TSDBConstants.TSDB_DATA_TYPE_NCHAR: {
+            case TSDBConstants.TSDB_DATA_TYPE_NCHAR:
+            case TSDB_DATA_TYPE_VARBINARY: {
```
Comment (sangshuduo) [inline src/main/java/com/taosdata/jdbc/TSDBResultSetRowData.java:95]: ditto

File: src/main/java/com/taosdata/jdbc/TSDBResultSetRowData.java [hunk 4]
```diff
@@ -426,9 +427,10 @@ public String getString(int col, int nativeType) {
                 return BigDecimal.valueOf(lowValue).add(BigDecimal.valueOf(Long.MAX_VALUE)).add(BigDecimal.valueOf(1)).toString();
             }
             case TSDBConstants.TSDB_DATA_TYPE_BINARY:
+            case TSDB_DATA_TYPE_VARBINARY:
```
Comment (sangshuduo) [inline src/main/java/com/taosdata/jdbc/TSDBResultSetRowData.java:430]: ditto

**Review by sangshuduo — COMMENTED** [review]
a few comments

File: src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java (hunk 1 above)
Comment (huolibo) [reply src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java:490]: yeah,  static import

**Pull request merged**